sympy/core/symbol.py

```diff
@@ -216,11 +221,17 @@ def __getstate__(self):
     @cacheit
     def sort_key(self, order=None):
         return self.class_key(), (
-            2, (str(self), self.dummy_index)), S.One.sort_key(), S.One
+            2, (str(self), Dummy._count.index(self.dummy_index))), S.One.sort_key(), S.One
```

Comment:
I'm slightly worried about performance here.  Probably `.index` is an O(n) search in the number of Dummy variables right?  If someone using a pattern with lots of Dummies in a tight loop, maybe its a problem.

I think I will record the `_count` in a dict keyed by `dummy_index` instead.  Then O(1) lookup here.